File: app/src/main/kotlin/tech/mmarca/openvitals/domain/query/NutritionPeriodData.kt
package tech.mmarca.openvitals.domain.query

import tech.mmarca.openvitals.domain.model.DailyMacros
import tech.mmarca.openvitals.domain.model.NutritionEntry

data class NutritionPeriodData(
    val dailyMacros: List<DailyMacros> = emptyList(),
    val previousDailyMacros: List<DailyMacros> = emptyList(),
    val baselineDailyMacros: List<DailyMacros> = emptyList(),
    val entries: List<NutritionEntry> = emptyList(),
)

File: lib/domain/query/nutrition_period_data.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../model/nutrition_models.dart';

part 'nutrition_period_data.freezed.dart';

@freezed
abstract class NutritionPeriodData with _$NutritionPeriodData {
  const factory NutritionPeriodData({
    @Default(<DailyMacros>[]) List<DailyMacros> dailyMacros,
    @Default(<DailyMacros>[]) List<DailyMacros> previousDailyMacros,
    @Default(<DailyMacros>[]) List<DailyMacros> baselineDailyMacros,
    @Default(<NutritionEntry>[]) List<NutritionEntry> entries,
  }) = _NutritionPeriodData;
}
